CHANGES.rst

```diff
@@ -1,6 +1,256 @@
 
+Release 6.1.0
+-------------
+
+Released: 2020-05-XX
+
+The Traits library is a foundational component of the Enthought Tool Suite. It
+provides observable, typed attributes for Python classes, making those classes
+suitable for event-driven dataflow programming and for immediate use as models
+for graphical user interfaces, like those provided by the TraitsUI library.
+
+Traits 6.1 is the latest feature release in the Traits 6 series, and contains
+several major improvements.
+
+Highlights of this release
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* A new :mod:`observation <traits.observation>` framework for observing traited
+  attributes and other observable objects has been introduced. This is intended
+  to provide a full replacement for the existing :func:`on_trait_change`
+  mechanism, and aims to fix a number of fundamental flaws in the design of
+  that mechanism. See the :ref:`observation <observe-notification>` section of
+  the user manual for an introduction to this framework.
+
+* New :class:`~traits.trait_list_object.TraitList`,
+  :class:`~traits.trait_dict_object.TraitDict` and
+  :class:`~traits.trait_set_object.TraitSet` classes have been added,
+  subclassing Python's built-in ``list``, ``dict`` and ``set`` (respectively).
+  Instances of these classes are observable objects in their own right, and
+  it's possible to attach observers to them directly. These classes were
+  primarily introduced to support the new observation framework, and are not
+  expected to be used directly. The API for these objects and their
+  notification system is provisional, and may change in a future Traits
+  release.
+
+* A new :class:`.Union` trait type has been added. This is intended as a simpler
+  replacement for the existing :class:`.Either` trait type. The :class:`.Either`
+  trait type may eventually be deprecated: users are encouraged to replace uses
+  of :class:`.Either` with :class:`.Union` where possible.
+
+* New :class:`.PrefixList`, :class:`.PrefixMap` and :class:`.Map` trait types
+  have been added. These replace the existing :class:`.TraitPrefixList`,
+  :class:`.TraitPrefixMap` and :class:`.TraitMap` subclasses of
+  :class:`.TraitHandler`, which are deprecated.
+
+* ``mypy`` typing stubs for the Traits library have been added in a separate
+  ``traits-stubs`` package, which will be released separately to PyPI. This
+  should help support Traits-using projects that want to make use of type
+  annotations and type checking.
+
+
+Notes on upgrading
+~~~~~~~~~~~~~~~~~~
+
+For the most part, Traits 6.1 should be entirely backwards compatible
+with Traits 6.0. However, there are a few things to be aware of.
+
+* Traits 6.1 expects TraitsUI 7.0 or later to be installed: a combination
+  of Traits 6.1 or later with TraitsUI 6.x or earlier will fail to
+  properly recognise :class:`View` class variables as TraitsUI views. If you
+  need this combination, a possible workaround is to use the
+  :meth:`default_traits_view` method.
```

Comment:
I tried but this workaround does not seem to work. That said, the sentence does say it is *possible*.  For my own understanding, under what situation can one use this as a workaround?

Reply:
I think the two of us have witnessed different failure modes: you're getting an exception, but I was seeing everything working except that the view brought up was the default traits view instead of the specified one. The workaround I describe is for the second case. I'll investigate both of these and see what I can find.

Reply:
Yep, the workaround won't work (either in master, or after your PR). This needs to be removed.

Reply:
Removed.